src/module/apps/sidebar/compendium-directory.ts

```diff
@@ -63,27 +63,24 @@ class CompendiumDirectoryPF2e extends CompendiumDirectory {
     protected override _getEntryContextOptions(): EntryContextOption[] {
         const options = super._getEntryContextOptions();
 
-        if (BUILD_MODE === "development" || game.settings.get("pf2e", "dataTools")) {
-            options.push({
-                name: "COMPENDIUM.Migrate",
-                icon: fontAwesomeIcon("crow").outerHTML,
-                condition: ($li) => {
-                    const compendium = game.packs.get($li.data("pack"), { strict: true });
-                    const actorOrItem =
-                        compendium.documentClass === CONFIG.Actor.documentClass ||
-                        compendium.documentClass === CONFIG.Item.documentClass;
-                    const isSystemCompendium = compendium.metadata.packageType === "system";
-                    return game.user.isGM && actorOrItem && !isSystemCompendium && !compendium.locked;
-                },
-                callback: async ($li) => {
-                    const compendium = game.packs.get($li.data("pack"), { strict: true }) as CompendiumCollection<
-                        ActorPF2e<null> | ItemPF2e<null>
-                    >;
-                    const runner = new MigrationRunner(MigrationList.constructFromVersion(null));
-                    runner.runCompendiumMigration(compendium);
-                },
-            });
-        }
+        options.push({
```

Comment:
Can this be restricted to GMs?

Reply:
It already is in the condition, unless you mean you want to move that out of the condition. Does getting your perms updated mid session force a refresh?